### PR TITLE
Add postgres-database-migration and ghost-database skills

### PR DESCRIPTION
**Pull request by cevian**
## Summary
- Add **postgres-database-migration** skill: DDL lock reference, safe migration patterns (add/drop/rename columns, type changes, NOT NULL, FK, indexes, PK redefinition), timeout strategies, transaction strategy, rollback planning, fork-based testing, and advanced considerations. Reference files for backfill strategies, validation queries, and a complete end-to-end example.
- Add **ghost-database** skill: Ghost CLI and MCP integration, Spaces vs dedicated instances, database lifecycle management (create, fork, pause, resume, delete), read-only access, and when to use Ghost.
- Update **postgres** router skill with new Migrations and Database Management sections, plus symlinked references for reliable path resolution.

## Test plan
- [ ] `./bun run validate-skills` passes for all skills
- [ ] Symlinks in `postgres/references/` resolve correctly
- [ ] `npx skills add timescale/pg-aiguide --list` shows the new skills

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: skills/postgres-database-migration/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+These complete in milliseconds regardless of table size. They only hold a brief `AccessExclusiveLock` for the catalog update, not for data rewriting.
+
+| Operation | Lock Level | Notes |
+|-----------|-----------|-------|
+| `ADD COLUMN` (nullable, no default) | `AccessExclusiveLock` (brief) | **Fast.** No table rewrite. Metadata-only change. |
+| `ADD COLUMN ... DEFAULT x` (PG 11+) | `AccessExclusiveLock` (brief) | **Fast.** Non-volatile defaults stored in catalog, not backfilled. |
+| `DROP COLUMN` | `AccessExclusiveLock` (brief) | **Fast.** Column marked invisible; space reclaimed by VACUUM over time. |
+| `SET DEFAULT` / `DROP DEFAULT` | `AccessExclusiveLock` (brief) | Metadata change only. Does not touch existing rows. |
+| `CREATE INDEX CONCURRENTLY` | `ShareUpdateExclusiveLock` | **Non-blocking.** Allows reads and writes during build. Slower than regular index creation. |
+| `DROP INDEX CONCURRENTLY` | None (blocking) | Waits for queries using the index to finish, then drops. |
```
Comment (jgpruitt) [inline skills/postgres-database-migration/SKILL.md]: What does `None (blocking)` mean? No blocking locks are taken?

File: skills/postgres-database-migration/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+-- Step 3: build unique index concurrently (non-blocking)
+CREATE UNIQUE INDEX CONCURRENTLY idx_orders_new_pkey
+    ON orders (new_id);
+
+-- Step 4: drop the old PK
+ALTER TABLE orders DROP CONSTRAINT orders_pkey;
+
+-- Step 5: add new PK using the existing index (instant — also implicitly adds NOT NULL)
+ALTER TABLE orders ADD CONSTRAINT orders_pkey
+    PRIMARY KEY USING INDEX idx_orders_new_pkey;
```
Comment (jgpruitt) [inline skills/postgres-database-migration/SKILL.md:236]: I think I'd wrap step 4 and 5 in a transaction so there's never a moment where the table has no primary key. Right?

Comment (jgpruitt) [reply skills/postgres-database-migration/SKILL.md:236]: But i guess that doesn't buy you anything because the unique index is already enforcing what you want.

File: skills/postgres-database-migration/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+    attempt INTEGER := 1;
+    success BOOLEAN := FALSE;
+BEGIN
+    WHILE attempt <= max_attempts AND NOT success LOOP
+        BEGIN
+            SET lock_timeout = '3s';
+            -- Replace with your DDL statement
+            ALTER TABLE orders ADD COLUMN tracking_number TEXT;
+            success := TRUE;
+            RAISE NOTICE 'DDL succeeded on attempt %', attempt;
```
Comment (jgpruitt) [inline skills/postgres-database-migration/SKILL.md:401]: Another pattern is to explicitly attempt to grab the lock with nowait. If you don't get it, sleep and retry.

File: skills/postgres-database-migration/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+        RAISE EXCEPTION 'DDL failed after % attempts', max_attempts;
+    END IF;
+END $$;
+```
+
+This prevents the migration from creating a pile-up of blocked queries behind it. Each attempt either succeeds quickly or gives up and lets normal traffic flow.
+
+## Fork-Based Migration Testing
+
+The safest way to test a migration is to run it against a copy of your actual database — same schema, same data, same edge cases. The only two providers that support fast database forking are [Neon](https://neon.tech) and [Ghost](https://ghost.build). Without database forking, you need to manually dump and restore your database, which can take a long time for large datasets.
```
Comment (jgpruitt) [inline skills/postgres-database-migration/SKILL.md:431]: It doesn't catch issues with concurrent database traffic. That's really hard to simulate. Probably worth noting.

File: skills/ghost-database/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+  **Trigger when user asks to:**
+  - Create a new PostgreSQL database quickly
+  - Manage database lifecycle in agent workflows
+  - Fork a database to test changes safely
+  - Create a database for agents
+  - Need many databases that are billed only when queried
+  - Get a connection string for a Ghost database
+  - Run SQL queries against a Ghost database
+  - Pause, resume, or delete Ghost databases
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]: We got rid of the ability to pause databases

File: skills/ghost-database/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+  - Set up the Ghost MCP server or CLI
+
+  **Keywords:** Ghost, ghost.build, database, create database, fork database, PostgreSQL, managed Postgres, MCP, agent database, connection string, ghost_create, ghost_fork, ghost_sql
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]: Including the MCP tool names (e.g. `ghost_create`, `ghost_fork`, `ghost_sql`) in the keywords feels a little strange to me. When would users or agents ever actually be looking for those specific MCP tools by name?

File: skills/ghost-database/SKILL.md [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+# Ghost Database Management
+
+Ghost is a managed PostgreSQL service designed for agents. Your account is organized into Spaces, each with unlimited databases and forks sharing a pool of compute hours (billed in 15-minute chunks when queries are executed). Free tier: 100 hours/month, 1TB storage.
+
+When you need continuous operation, promote any database to a **dedicated instance** — a separately billed, always-on database.
+
+Additional features:
+- **CLI and MCP native:** create and query databases from the terminal or any MCP-compatible agent
+- **Instant forking:** full database copies in seconds for safe experimentation
+- **Read-only access:** connect with `--read-only` for safe agentic work against production data
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]: I think this is slightly misleading.

The `--read-only` flag can be passed to `ghost connect` (to get a connection string that uses read-only mode) or `ghost psql` (to connect via psql in read-only mode), but those are mostly for _humans_.

For agents, you set read-only mode with `ghost config set read_only true`. After that, all MCP tools are locked into read-only mode - i.e. queries executed with the `ghost_sql` tool are executed in read-only mode, and the MCP tools that modify resources are blocked (i.e. `ghost_delete`, `ghost_password`, `ghost_rename` are all blocked in read-only mode).

Might be worth explaining this in more detail below (it's a pretty useful feature that isn't well-documented anywhere else).

File: skills/ghost-database/SKILL.md [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+
+Website: https://ghost.build
+GitHub: https://github.com/timescale/ghost
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]: The repo is not public (at least not yet). Probably shouldn't link to it here.

File: skills/ghost-database/SKILL.md [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+
+## Installation
+
+```bash
+curl -fsSL https://install.ghost.build | sh
+```
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md:47]: We support [other installation methods](https://github.com/timescale/ghost?tab=readme-ov-file#installation) too, such as a PowerShell script for Windows, and installing Linux packages from Packagecloud. Probably worth mentioning them for the sake of completeness.

File: skills/ghost-database/SKILL.md [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+## Core Workflows
+
+### Create and Query a Database
+
+```bash
+# Create a database
+ghost create --name my-app-db
+
+# Run SQL directly
+ghost sql my-app-db "CREATE TABLE users (id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY, email TEXT NOT NULL UNIQUE, created_at TIMESTAMPTZ NOT NULL DEFAULT now())"
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]: Commands take the database ID, not the name. This example is using the name specified in the `ghost create` command above, which is incorrect.

The same is true for all other `ghost` commands, as well. Need to make sure it's correct everywhere in this doc.

File: skills/ghost-database/SKILL.md [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+```
+
+### Pause and Resume
+
+Pause databases you're not actively using to conserve compute hours. Storage is retained.
+
+```bash
+ghost pause my-app-db
+ghost resume my-app-db --wait
+```
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]: As mentioned above, we removed the `ghost pause` command.

We should probably explain when and why databases get paused instead (I think it's after 30 days of idle time now?), and just explain how to resume them with `ghost resume`.

File: skills/ghost-database/SKILL.md [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+| `ghost schema` | Display database schema |
+| `ghost list` | List all databases (`--json`, `--yaml`) |
+| `ghost status` | Show space usage |
+| `ghost pause` | Pause a running database |
+| `ghost resume` | Resume a paused database (`--wait`) |
+| `ghost password` | Reset password (`--generate` for auto-generated) |
+| `ghost rename` | Rename a database |
+| `ghost logs` | View database logs |
+| `ghost psql` | Open interactive psql session (`--read-only`) |
+| `ghost login` | Authenticate with GitHub OAuth (`--headless` for CI) |
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]: This is not a complete list of commands. It's missing things like `ghost api-key`, `ghost config`, `ghost logout`, `ghost feedback`, `ghost mcp`, `ghost version`, and potentially others. We should probably audit this to make sure it's complete. You can easily get the full list from `ghost --help`.

File: skills/ghost-database/SKILL.md [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+
+## MCP Integration
+
+The Ghost MCP server gives agents full database lifecycle control — create, fork, query, inspect, pause, resume, and delete databases without human intervention.
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]:
```suggestion
The Ghost MCP server gives agents full database lifecycle control — create, fork, query, inspect, resume, and delete databases without human intervention.
```

File: skills/ghost-database/SKILL.md [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+| `ghost_list` | List all databases |
+| `ghost_status` | Show space usage |
+| `ghost_pause` | Pause a database |
+| `ghost_resume` | Resume a database |
+| `ghost_password` | Reset password |
+| `ghost_rename` | Rename a database |
+| `ghost_logs` | View logs |
+| `ghost_login` | Authenticate with GitHub OAuth |
+| `ghost_feedback` | Submit feedback |
+| `search_docs` | Search Ghost and Postgres documentation |
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]: There is also `view_skill` (proxied from pg-aiguide)

**Review by nathanjcochran — COMMENTED** [review]
I only reviewed the Ghost skill. Largely looks good to me, but I left a handful of comments.

It's also worth noting that the skill mentioned dedicated databases but they aren't technically support yet, so we might want to wait to merge this until that work is done? And at that point, we'll want to add info about the additional commands for creating/forking dedicated instances, as well as the payment-related commands (i.e. `ghost payment`).

Additionally, we might want to talk a bit about API keys (i.e. the `ghost api-key` commands) and how to use them to access the API programatically? Seems like it could be useful in some cases (e.g. if building an app that integrates with Ghost directly, or if setting up a CI pipeline that forks a database to run tests against).

File: skills/postgres-database-migration/references/validation-queries.md [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+-- Table size and row count (estimate for planning)
+SELECT
+    pg_size_pretty(pg_total_relation_size('orders')) AS total_size,
+    pg_size_pretty(pg_relation_size('orders')) AS data_size,
+    reltuples::BIGINT AS estimated_rows
+FROM pg_class
+WHERE relname = 'orders';
+
+-- Estimate backfill time: run a small batch and extrapolate
+EXPLAIN ANALYZE
```
Comment (jgpruitt) [inline skills/postgres-database-migration/references/validation-queries.md:77]: Should we warn that ANALYZE will cause it to actually update rows?

**Review by jgpruitt — APPROVED** [review]

**Comment by cevian** [comment]
Re: dedicated instances — leaving them in for now since dedicated instance support should be merged soon. We can update with the additional commands (ghost payment, etc.) when that lands.

File: skills/ghost-database/SKILL.md (hunk 7 above)
Comment (cevian) [reply skills/ghost-database/SKILL.md]: can you please check the wording now?

File: skills/ghost-database/SKILL.md [hunk 16]
```diff
@@ -0,0 +1,209 @@
...
+
+# Ghost Database Management
+
+Ghost is a managed PostgreSQL service designed for agents. Your account is organized into Spaces, each with unlimited databases and forks sharing a pool of compute hours (billed in 15-minute chunks when queries are executed). Free tier: 100 hours/month, 1TB storage.
+
+When you need continuous operation, promote any database to a **dedicated instance** — a separately billed, always-on database.
+
+Additional features:
+- **CLI and MCP native:** create and query databases from the terminal or any MCP-compatible agent
+- **Instant forking:** full database copies in seconds for safe experimentation
```
Comment (murrayju) [inline skills/ghost-database/SKILL.md:33]: I thought we were avoiding "instant", since few people consider 90s to be instantaneous.

File: skills/ghost-database/SKILL.md [hunk 17]
```diff
@@ -0,0 +1,209 @@
...
+curl -s https://packagecloud.io/install/repositories/timescale/ghost/script.deb.sh | sudo os=any dist=any bash
+sudo apt-get install ghost
+```
+
+### Red Hat/Fedora
+
+```bash
+curl -s https://packagecloud.io/install/repositories/timescale/ghost/script.rpm.sh | sudo os=rpm_any dist=rpm_any bash
+sudo yum install ghost
+```
```
Comment (murrayju) [inline skills/ghost-database/SKILL.md:67]: I think this skill could benefit from some progressive disclosure, and this section in particular seems like it'd _usually_ be a waste of tokens. You only need to _install_ ghost once.

File: skills/ghost-database/SKILL.md [hunk 18]
```diff
@@ -0,0 +1,209 @@
...
+ghost create --name my-app-db
+
+# Run SQL directly (use the database ID)
+ghost sql abc123 "CREATE TABLE users (id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY, email TEXT NOT NULL UNIQUE, created_at TIMESTAMPTZ NOT NULL DEFAULT now())"
+
+# Query it
+ghost sql abc123 "SELECT * FROM users"
+
+# Open interactive psql session
+ghost psql abc123
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md:103]: We might want to say that this requires `psql` to be installed and in your `PATH`

File: skills/ghost-database/SKILL.md [hunk 19]
```diff
@@ -0,0 +1,209 @@
...
+  **Trigger when user asks to:**
+  - Create a new PostgreSQL database quickly
+  - Manage database lifecycle in agent workflows
+  - Fork a database to test changes safely
+  - Create a database for agents
+  - Need many databases that are billed only when queried
+  - Get a connection string for a Ghost database
+  - Run SQL queries against a Ghost database
+  - Resume or delete Ghost databases
+  - Set up the Ghost MCP server or CLI
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md:15]: Given that the Ghost MCP currently proxies the pg-aiguide MCP tools (including `view_skill`), are we at all concerned that this skill could create confusion for agents that are already using the Ghost MCP?

Like, if a user asks to fork a database, how will the agent decide between using this skill (which mostly recommends using `ghost` CLI commands directly, which is not really what we want if the MCP is installed) or just calling the `ghost_fork` tool? Maybe it would use the skill first and then still decide to call the MCP tool afterwards... but even in that case, using the skill would just be extra wasted tokens (the Ghost MCP is already self-describing, so if it's already installed, I don't think there's much more that this skill would help with, beyond maybe explaining some of the less well-documented features).

It seems like this skill is _mostly_ geared towards setting up Ghost the first time and understanding its capabilities (e.g. when evaluating whether to use it). But if Ghost is already installed (as MCP server), I feel like this skill could potentially be counter-productive.

Any ideas for what we should do about that? We could potentially just filter this skill out when we proxy the `view_skill` tool, but we don't do anything like that currently (we proxy the tools as-is), and in any case, it wouldn't work for pre-existing versions of Ghost that are already in the wild.

File: skills/ghost-database/SKILL.md (hunk 17 above)
Comment (nathanjcochran) [reply skills/ghost-database/SKILL.md:67]: True. I think maybe we should try to decide whether this skill is mostly for getting Ghost set up the first time (in which case maybe listing all of the installation methods here is fine), or if we think it's useful as a continual reference for how to use Ghost. Personally, I'm kind of skeptical that it's going to be helpful once Ghost is already installed (since the MCP is self-describing anyways, and this skill would just be a conflicting source of information). See my other comment about that. I think maybe we should try to narrow in on exactly how/when we want this skill to be used, because I'm a little concerned that this skill could end up taking priority over using the Ghost MCP tools directly.

File: skills/ghost-database/SKILL.md (hunk 7 above)
Comment (nathanjcochran) [reply skills/ghost-database/SKILL.md]: Looks good! Thanks for making those changes 👍

File: skills/ghost-database/SKILL.md (hunk 17 above)
Comment (murrayju) [reply skills/ghost-database/SKILL.md:67]: Yeah, that's fair. I agree this skill is redundant with the MCP tools, and the skill description over-promises on what it is for in a way that could be problematic.

I kinda think there should be a "choosing a db provider" skill, and then possibly an "installing ghost" sub-page.

File: skills/ghost-database/SKILL.md [hunk 20]
```diff
@@ -0,0 +1,259 @@
+---
+name: ghost-database
+description: |
+  Use this skill for persistent PostgreSQL storage in agent workflows — spin up databases on demand, test schema migrations safely on forks, and pass databases as assets between agents using Ghost.
+
+  **Trigger when user asks to:**
+  - Create a new PostgreSQL database quickly
```
Comment (murrayju) [inline skills/ghost-database/SKILL.md:7]: Do we need "quickly" here?

File: skills/ghost-database/SKILL.md [hunk 21]
```diff
@@ -0,0 +1,259 @@
...
+
+# Ghost Database Management
+
+Ghost is a managed PostgreSQL service designed for agents. Your account is organized into Spaces, each with unlimited on-demand databases and forks sharing a pool of compute hours (billed in 15-minute chunks when queries are executed). Free tier: 100 hours/month, 1TB storage.
+
+When you need continuous operation, promote any database to a **dedicated instance** — a separately billed, always-on database.
+
+Additional features:
+- **CLI and MCP native:** create and query databases from the terminal or any MCP-compatible agent
+- **Instant forking:** full database copies in seconds for safe experimentation
```
Comment (murrayju) [inline skills/ghost-database/SKILL.md:33]:
```suggestion
- **Zero-copy forking:** full database copies in seconds for safe experimentation
```

**Review by murrayju — APPROVED** [review]
Some of my comments are still unaddressed, and I do think this could be better organized and use progressive disclosure better, to better provide the right context for the right tasks (where the ghost skill here is really a mix of several use cases).

But if you want to start here, I don't really see anything that's a _problem_.

File: skills/ghost-database/SKILL.md [hunk 22]
```diff
@@ -0,0 +1,259 @@
...
+```
+
+## Getting Started
+
+**CLI**
+```bash
+ghost login                     # Authenticate with GitHub
+ghost create                    # Create a new database (returns an ID, e.g. abc123)
+ghost list                      # List all databases with their IDs
+ghost connect <id>              # Get connection string
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]:
```suggestion
ghost connect <name-or-id>              # Get connection string
```

File: skills/ghost-database/SKILL.md [hunk 23]
```diff
@@ -0,0 +1,259 @@
...
+```
+
+**MCP**
+```
+ghost_login()                   // Authenticate with GitHub
+ghost_create({ name: "my-db" }) // → returns { id: "abc123", ... }
+ghost_list()                    // List all databases with their IDs
+ghost_connect({ id: "abc123" }) // Get connection string
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]:
```suggestion
ghost_connect({ name_or_id: "abc123" }) // Get connection string
```

File: skills/ghost-database/SKILL.md [hunk 24]
```diff
@@ -0,0 +1,259 @@
...
+```
+
+## Core Workflows
+
+### Create and Query a Database
+
+**CLI**
+```bash
+# Create a database (returns an ID like abc123)
+ghost create --name my-app-db
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]:
```suggestion
ghost create my-app-db
```

File: skills/ghost-database/SKILL.md [hunk 25]
```diff
@@ -0,0 +1,259 @@
...
+
+# Run SQL directly (use the database ID)
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]:
```suggestion
# Run SQL directly
```

File: skills/ghost-database/SKILL.md [hunk 26]
```diff
@@ -0,0 +1,259 @@
...
+# Open interactive psql session
+ghost psql abc123
+```
+
+**MCP**
+```
+ghost_create({ name: "my-app-db" })
+// → returns { id: "abc123", ... }
+
+ghost_sql({ id: "abc123", query: "CREATE TABLE users (id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY, email TEXT NOT NULL UNIQUE, created_at TIMESTAMPTZ NOT NULL DEFAULT now())" })
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]:
```suggestion
ghost_sql({ name_or_id: "abc123", query: "CREATE TABLE users (id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY, email TEXT NOT NULL UNIQUE, created_at TIMESTAMPTZ NOT NULL DEFAULT now())" })
```

File: skills/ghost-database/SKILL.md [hunk 27]
```diff
@@ -0,0 +1,259 @@
...
+
+ghost_sql({ id: "abc123", query: "SELECT * FROM users" })
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]:
```suggestion
ghost_sql({ name_or_id: "abc123", query: "SELECT * FROM users" })
```

File: skills/ghost-database/SKILL.md [hunk 28]
```diff
@@ -0,0 +1,259 @@
...
+### Fork for Safe Experimentation
+
+Forking creates a full copy of your database in seconds — same schema, same data. Use forks to test migrations, experiment with schema changes, or let agents explore without risk to your working database. You can fork a dedicated instance into an on-demand instance — useful for testing against a production copy without paying for always-on compute.
+
+For a complete migration testing workflow using forks — including pre/post validation queries and rollback planning — see the `postgres-database-migration` skill.
+
+**CLI**
+```bash
+# Fork a database (returns the fork's ID, e.g. def456)
+ghost fork abc123 --name my-app-db-experiment
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]:
```suggestion
ghost fork abc123 my-app-db-experiment
```

File: skills/ghost-database/SKILL.md [hunk 29]
```diff
@@ -0,0 +1,259 @@
...
+
+# Test changes on the fork
+ghost sql def456 "ALTER TABLE users ADD COLUMN role TEXT NOT NULL DEFAULT 'user'"
+
+# If it worked: apply to original
+ghost sql abc123 "ALTER TABLE users ADD COLUMN role TEXT NOT NULL DEFAULT 'user'"
+
+# If it failed: delete the fork, original is untouched
+ghost delete def456 --confirm
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]: Since the commands take a name _or_ ID now, you could update these to use the name from above (e.g. `my-app-db-experiment`) instead of the ID (e.g. `def456`)

File: skills/ghost-database/SKILL.md [hunk 30]
```diff
@@ -0,0 +1,259 @@
...
+```
+
+**MCP**
+```
+ghost_fork({ id: "abc123", name: "my-app-db-experiment" })
+// → returns { id: "def456", ... }
+
+ghost_sql({ id: "def456", query: "ALTER TABLE users ADD COLUMN role TEXT NOT NULL DEFAULT 'user'" })
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]:
```suggestion
ghost_sql({ name_or_id: "def456", query: "ALTER TABLE users ADD COLUMN role TEXT NOT NULL DEFAULT 'user'" })
```

File: skills/ghost-database/SKILL.md [hunk 31]
```diff
@@ -0,0 +1,259 @@
...
+# If it worked: apply to original
+ghost sql abc123 "ALTER TABLE users ADD COLUMN role TEXT NOT NULL DEFAULT 'user'"
+
+# If it failed: delete the fork, original is untouched
+ghost delete def456 --confirm
+```
+
+**MCP**
+```
+ghost_fork({ id: "abc123", name: "my-app-db-experiment" })
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]:
```suggestion
ghost_fork({ name_or_: "abc123", name: "my-app-db-experiment" })
```

File: skills/ghost-database/SKILL.md [hunk 32]
```diff
@@ -0,0 +1,259 @@
...
+// → returns { id: "def456", ... }
+
+ghost_sql({ id: "def456", query: "ALTER TABLE users ADD COLUMN role TEXT NOT NULL DEFAULT 'user'" })
+
+// If it worked: apply to original
+ghost_sql({ id: "abc123", query: "ALTER TABLE users ADD COLUMN role TEXT NOT NULL DEFAULT 'user'" })
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]:
```suggestion
ghost_sql({ name_or_id: "abc123", query: "ALTER TABLE users ADD COLUMN role TEXT NOT NULL DEFAULT 'user'" })
```

File: skills/ghost-database/SKILL.md [hunk 33]
```diff
@@ -0,0 +1,259 @@
...
+
+// If it failed: delete the fork, original is untouched
+ghost_delete({ id: "def456" })
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]:
```suggestion
ghost_delete({ name_or_id: "def456" })
```

File: skills/ghost-database/SKILL.md [hunk 34]
```diff
@@ -0,0 +1,259 @@
...
+Databases automatically pause after 30 days of idle time to conserve compute hours. Storage is retained. Resume a paused database when you need it again:
+
+**CLI**
+```bash
+ghost resume abc123 --wait
+```
+
+**MCP**
+```
+ghost_resume({ id: "abc123" })
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]:
```suggestion
ghost_resume({ name_or_id: "abc123" })
```

File: skills/ghost-database/SKILL.md [hunk 35]
```diff
@@ -0,0 +1,259 @@
...
+### Inspect Schema
+
+**CLI**
+```bash
+ghost schema abc123
+```
+
+**MCP**
+```
+ghost_schema({ id: "abc123" })
```
Comment (nathanjcochran) [inline skills/ghost-database/SKILL.md]:
```suggestion
ghost_schema({ name_or_id: "abc123" })
```

**Review by nathanjcochran — APPROVED** [review]
Left a bunch of small suggestions relating to IDs vs. names, since we've updated how that works since this was originally written.

We should definitely remember to update both Ghost and Tiger CLI (which both proxy the MCP tools/skills from this repo) to exclude the new Ghost skill. Otherwise, I'm concerned that it could cause conflicting advice to agents, and cause them to get confused or do the wrong thing (e.g. try to use Ghost instead of the Tiger CLI/MCP).
